### ISSUE TITLE
Support for d3 4.x

### ISSUE THREAD

**Issue by niemyjski** [shutterstock/rickshaw#544]
What is the support for d3 4.x like? Any known issues?

**Comment by niemyjski** [comment]
Has anyone ran into issues with this?

**Comment by guillaumegarcia13** [comment]
Yep, I did.
Run stacked chart and faced d3.layout.stack() error because d3.layout got undefined. (read somewhere it was split into a separate file for d3 4.x)
Reverted back to latest 3.x branch and it solved my issue.

**Comment by Flaburgan** [comment]
I'm interested about this, I'm new to d3.js and [the examples](http://code.shutterstock.com/rickshaw/tutorial/introduction.html) talk about d3.layout and I can't find this file. Even if there is a "recent" 1.6 version of rickshaw, nothing looks updated on the website...

**Comment by niemyjski** [comment]
I wish a core contributor would respond, the lack of response kind of tells me this project is dead :(. I wish I was better at charting/visuals and I had more time, then I would contribute in the time in which I don't have :(

**Comment by butterflyhug** [comment]
I'm comprehensively updating the code and examples to work with d3 v4; my team uses Rickshaw and we can't keep postponing the d3 version upgrade.  I'll push my code to GitHub this weekend, as soon as I clean up my commit history for public consumption.

Would anyone @shutterstock be interested in merging a PR to update to d3 v4?  I can see from #517 that y'all have largely stepped back from maintaining this.  I'm happy to send a PR with my code and to help keep the code healthy if you want contributions.  Or if you'd rather keep this repo stable, I'm also happy to keep my fork running for my team (and anyone else who would prefer to use that).

**Comment by niemyjski** [comment]
@butterflyhug any updates on this?

**Comment by butterflyhug** [comment]
I've got code [on my fork](https://github.com/butterflyhug/rickshaw).  Note that my current default branch is `d3v4` -- **not** `master`.

I've added a section at the top of the readme to note a couple minor issues that I noticed from testing my code with the provided examples.  I've also tested with the existing unit tests and my team's own codebase, both of which seem to be working flawlessly.  I'm sure that there are a few features and/or configurations that haven't been covered by any of these testing methods, and those things may be broken.

I'm still happy to send a PR with my changes if @shutterstock is interested.  In the meantime, I'm happy to take PRs to address remaining issues on my fork.  If we get to a point where there are no known issues and I still haven't heard any official interest in supporting D3 v4, then I'll think about our options for independently publishing releases and etc.

**Comment by RichardLitt** [comment]
@butterflyhug:

> Would anyone @shutterstock be interested in merging a PR to update to d3 v4? I can see from #517 that y'all have largely stepped back from maintaining this. I'm happy to send a PR with my code and to help keep the code healthy if you want contributions. Or if you'd rather keep this repo stable, I'm also happy to keep my fork running for my team (and anyone else who would prefer to use that)

Yes, I would be more than happy to look at a PR for this update, and I would love to have help maintaining this repository. PR away!

> I'm still happy to send a PR with my changes if @shutterstock is interested.

:wave:

**Comment by socketbox** [comment]
Any fix should include replacing deprecated calls to `d3.scale.linear()` with `d3.scaleLinear()`

**Comment by Justus-Maier** [comment]
What's missing here? Response from @shutterstock?

**Comment by niemyjski** [comment]
Updated title and issue to 5.x since we are now two majors behind.
